### ISSUE TITLE
Bug for read -> write -> write sequence in transaction

### ISSUE THREAD

**Issue by mweberUKL** [AntidoteDB/antidote#346]
Write operations performed inside an interactive transaction should order the updates according to the session order. This works for update-only transactions. But performing a read before-hand leads to a confusion of the update-order.

Here an example of the problem in Java:

```
MVRegisterKey<String> regKey = Key.multiValueRegister("testReg");
InteractiveTransaction txn = antidoteClient.startTransaction();

// this read is a show killer!
List<String> initialRead = bucket.read(txn, regKey);

bucket.update(txn, regKey.assign("test1"));

bucket.update(txn, regKey.assign("test2"));

List<String> valInsideTx2 = bucket.read(txn, regKey);

txn.commitTransaction();
List<String> valOutsideTx = bucket.read(antidoteClient.noTransaction(), regKey);

assertEquals(singletonList("test2"), valInsideTx2);
assertEquals(singletonList("test2"), valOutsideTx);
```

When commenting out the marked line, the test runs through. But reading the value before the update makes the test fail because both "test1" and "test2" are returned as result of the multi-value register.

**Comment by peterzeller** [comment]
Quick and dirty fix:
Change `clocksi_interactive_coord` line 488 and don't update the ReadSet.

@deepthidevaki Can you have a look at it? Probably it is better to also update the `internal_read_set` after updates.

**Comment by deepthidevaki** [comment]
fixed in pull request #353
